logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/IncrementalSyncManager.kt

```diff
@@ -61,45 +61,51 @@ internal class IncrementalSyncManager(
     @OptIn(ExperimentalCoroutinesApi::class)
     private val eventProcessingDispatcher = kaliumDispatcher.default.limitedParallelism(1)
 
-    private val coroutineExceptionHandler = CoroutineExceptionHandler { _, throwable ->
-        when (throwable) {
-            is CancellationException -> {
-                kaliumLogger.withFeatureId(SYNC).i("Sync job was cancelled")
-                incrementalSyncRepository.updateIncrementalSyncState(IncrementalSyncStatus.Pending)
-            }
-
-            is KaliumSyncException -> {
-                kaliumLogger.withFeatureId(SYNC).i("SyncException during events processing", throwable)
-                incrementalSyncRepository.updateIncrementalSyncState(IncrementalSyncStatus.Failed(throwable.coreFailureCause))
-            }
-
-            else -> {
-                kaliumLogger.withFeatureId(SYNC).i("Sync job failed due to unknown reason", throwable)
-                incrementalSyncRepository.updateIncrementalSyncState(IncrementalSyncStatus.Failed(CoreFailure.Unknown(throwable)))
-            }
-            // TODO: Trigger retry depending on failure
+    private val coroutineExceptionHandler = SyncExceptionHandler({
+        incrementalSyncRepository.updateIncrementalSyncState(IncrementalSyncStatus.Pending)
+    }, {
+        incrementalSyncRepository.updateIncrementalSyncState(IncrementalSyncStatus.Failed(it))
+        syncScope.launch {
+            delay(RETRY_DELAY)
+            startMonitoringForSync()
```

Comment:
Is this intended? we are starting the monitoring for sync here in the `coroutineExceptionHandler` and in the init block.

Reply:
This PR contains the changes from the retry mechanism.

This was added there.

The idea is that when an exception is thrown, the monitoring stops, and that handler is called. Depending on the exception, a simple delay is added and the monitoring starts again.